pkg/kubelet/cm/memorymanager/policy_static.go

```diff
@@ -83,6 +86,46 @@ func (p *staticPolicy) Start(s state.State) error {
 	return nil
 }
 
+func (p *staticPolicy) updateMemoryToReuse(pod *v1.Pod, container *v1.Container, blocks []state.Block){
+	// This will flush reusable resources for a pod after all containers have been allocated
+	// for podUID := range p.memoryToReuse {
+	// 	if podUID != string(pod.UID) {
+	// 		delete(p.memoryToReuse, podUID)
+	// 	}
+	// }
```

Comment:
Commented right now.

The same logic is applied in CPU Manager. It means that reusable for a pod will be deleted when all containers have been allocated resources and new pod is being allocated right now.

The problem I see here: what happens when after a couple of hours and few other pods placed a container crashes. (there may be a few seconds, minutes of BackOff), shouldn't it use the reusables? Can there be a racing for that memory with other containers? 

Here important may be a sequence (RemoveContainer(), AddContainer(), etc) when container crashes to understand that.

Reply:
That's right. Thanks for writing it down. I uncommented that part and deleted logic from `RemoveContainer()`